File: 898/Solution.java
class Solution {
    public int subarrayBitwiseORs(int[] arr) {
        Set<Integer> res = new HashSet<>();
        Set<Integer> cur = new HashSet<>();
        for (int elem : arr){
            Set<Integer> next = new HashSet<>();
            next.add(elem);
            for (int data : cur){
                next.add(data | elem);
            }
            res.addAll(next);
            cur = next;
        }
        return res.size();
    }
}

File: 898/solution.py
class Solution:
    def subarrayBitwiseORs(self, arr: List[int]) -> int:
        unique = set()
        temp = set()
        for elem in arr:
            temp = {u | elem for u in temp}
            temp.add(elem)
            unique.update(temp)
        return len(unique)